pretrain/openwebtext.py

```diff
@@ -101,7 +101,7 @@ def main(fabric: L.Fabric, resume: Union[bool, Path]) -> None:
     state = {"model": model, "optimizer": optimizer, "hparams": hparams, "iter_num": 0, "step_count": 0}
 
     if resume is True:
-        resume = sorted(out_dir.glob("*.pth"))[-1]
+        resume = sorted(out_dir.glob("*.pth"), key=lambda x: int(x.name.split("-")))[-1]
```

Comment:
`str.split()` returns a list. We can't call `int()` on a list. Perhaps it should be
```suggestion
        resume = sorted(out_dir.glob("*.pth"), key=lambda x: int(x.name.split("-")[-1]))[-1]
```

Reply:
Oh yea, thanks for the catch! I think it should be `resume = sorted(out_dir.glob("*.pth"), key=lambda x: int(x.name.split("-")[1]))[-1]` instead since the `iter_num` will be in the second index.

https://github.com/Lightning-AI/lit-gpt/blob/cf91ad2711632c7b11abb33ef235a81dfb381189/pretrain/redpajama.py#L220